File: src/main/java/io/github/alexiscomete/lapinousecond/resources/ResourceManager.java
package io.github.alexiscomete.lapinousecond.resources;

import java.util.Collection;
import java.util.HashMap;

public class ResourceManager {

    private final Resource resource;
    private int quantity;

    public ResourceManager(Resource resource, int quantity) {
        this.resource = resource;
        this.quantity = quantity;
    }

    public Resource getResource() {
        return resource;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public static HashMap<Resource, ResourceManager> stringToArray(String str) {
        HashMap<Resource, ResourceManager> resourceManagers = new HashMap<>();
        if (str == null || str.equals("")) {
            return resourceManagers;
        }
        String[] strings = str.split(" ");
        for (String s :
                strings) {
            String[] elements = s.split(":");
            if (elements.length > 1) {
                resourceManagers.put(Resource.valueOf(elements[0]) , new ResourceManager(Resource.valueOf(elements[0]), Integer.parseInt(elements[1])));
            }
        }
        return resourceManagers;
    }

    public static String toString(Collection<ResourceManager> resourceManagers) {
        StringBuilder stringBuilder = new StringBuilder();
        for (ResourceManager re :
                resourceManagers) {
            stringBuilder.append(re.toString()).append(" ");
        }
        return stringBuilder.toString();
    }

    @Override
    public String toString() {
        return resource.getProgName() + ":" + quantity;
    }
}

File: src/main/java/io/github/alexiscomete/lapinousecond/resources/ResourceManager.kt
package io.github.alexiscomete.lapinousecond.resources

class ResourceManager(val resource: Resource, var quantity: Int) {

    override fun toString(): String {
        // pour sauvegarder dans la base de données
        return resource.progName + ":" + quantity
    }

    companion object {
        fun stringToArray(str: String?): HashMap<Resource, ResourceManager> {
            // la futur liste de toutes les ressources du joueur
            val resourceManagers = HashMap<Resource, ResourceManager>()
            if (str == null || str == "") {
                return resourceManagers
            }
            // pn sépare chaque ressource
            val strings = str.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            // comme vu précédemment le nom est suivi de : et de la quabtité
            for (s in strings) {
                val elements = s.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                if (elements.size > 1) {
                    resourceManagers[Resource.valueOf(elements[0])] = ResourceManager(
                        Resource.valueOf(
                            elements[0]
                        ), elements[1].toInt()
                    )
                }
            }
            return resourceManagers
        }

        fun toString(resourceManagers: Collection<ResourceManager>): String {
            val stringBuilder = StringBuilder()
            for (re in resourceManagers) {
                stringBuilder.append(re.toString()).append(" ")
            }
            return stringBuilder.toString()
        }
    }
}
